src/test/java/com/slisenko/education/xml/XMLParserTest.java

```diff
@@ -49,165 +49,198 @@ public void parseStr() {
         List<Student> actual = parser.parseStr(badXml);
-        //List<Student> actual = parser.parseStr(strXml);
 
         Assert.assertEquals(expected, actual);
     }
 
     @Test
     public void parseStr_BadTableOpenTag() {
 
-        String tag = "<students>";
-        int index = strXml.indexOf(tag);
-        String badXml = strXml.substring(0,  index) + "<????>" +
-                strXml.substring(index + tag.length());
-        /*
-        String badXml = strXml.substring(0,  index) +
-                strXml.substring(index + tag.length());
-        */
+        // Bad tag <students> -> <????>
+        String badXml =
+                "<????>" +
+                    "<student>" +
+                        "<name>Name3</name><surname>Surname3</surname><year>1993</year><faculty>Faculty3</faculty><course>3</course>" +
+                    "</student>" +
+                    "<student>" +
+                        "<name>Name2</name><surname>Surname2</surname><year>1992</year><faculty>Faculty2</faculty><course>2</course>" +
+                    "</student>" +
+                    "<student>" +
+                        "<name>Name1</name><surname>Surname1</surname><year>1991</year><faculty>Faculty1</faculty><course>1</course>" +
+                    "</student>" +
+                "</students>";
+
         XMLParser parser = new XMLParser();
         List<Student> list = parser.parseStr(badXml);
-        //List<Student> list = parser.parseStr(strXml);
 
-        Assert.assertTrue(list.isEmpty());
+        if (!list.isEmpty()) {
+            Assert.fail();
+        }
     }
 
     @Test
     public void parseStr_BadTableCloseTag() {
 
-        String tag = "</students>";
-        int index = strXml.indexOf(tag);
-        /*
-        String badXml = strXml.substring(0,  index) + "</????>" +
-                strXml.substring(index + tag.length());
-        */
-        String badXml = strXml.substring(0,  index) +
-                strXml.substring(index + tag.length());
+        // Absent tag </students>
+        String badXml =
+                "<students>" +
+                    "<student>" +
+                        "<name>Name3</name><surname>Surname3</surname><year>1993</year><faculty>Faculty3</faculty><course>3</course>" +
+                    "</student>" +
+                    "<student>" +
+                        "<name>Name2</name><surname>Surname2</surname><year>1992</year><faculty>Faculty2</faculty><course>2</course>" +
+                    "</student>" +
+                    "<student>" +
+                        "<name>Name1</name><surname>Surname1</surname><year>1991</year><faculty>Faculty1</faculty><course>1</course>" +
+                    "</student>" +
+                "";
 
         XMLParser parser = new XMLParser();
         List<Student> list = parser.parseStr(badXml);
-        //List<Student> list = parser.parseStr(strXml);
 
-        //Assert.assertTrue(list.isEmpty());
         if (!list.isEmpty()) {
```

Comment:
use assertFalse(list.isEmpty())

Reply:
ok, but used assertTrue(list.isEmpty()) unstead of assertFalse(list.isEmpty())

Reply:
ok, sure, let's do assertTrue(list.isEmpty())